test/unit/core.js

```diff
@@ -286,13 +286,19 @@ QUnit.test( "type for `Symbol`", function( assert ) {
 });
 
 QUnit.asyncTest( "isPlainObject", function( assert ) {
-	assert.expect( 15 );
+	assert.expect( 17 );
 
-	var pass, iframe, doc,
+	var pass, iframe, doc, parentObj, childObj,
 		fn = function() {};
 
 	// The use case that we want to match
 	assert.ok( jQuery.isPlainObject( {} ), "{}" );
+	assert.ok( jQuery.isPlainObject( new Object() ), "new Object" ); // jshint ignore:line
+
+	parentObj = { foo: "bar" };
+	childObj = Object.assign( Object.create( parentObj ), { bar: "foo" } );
```

Comment:
Use case was presented with `Object.assign` so it seems thorough to check with it, but test without it wouldn't hurt either, i for one, not 100% if `Object.assign` doesn't change behaviour of extend object, so i would leave it, just to be sure.